corehq/util/couch_doc_processor.py

```diff
@@ -29,162 +52,34 @@ class ResumableDocsByTypeIterator(object):
-
-        args.update(
-            view_name='all_docs/by_doc_type',
-            chunk_size=chunk_size,
-            event_handler=event_handler,
-            include_docs=True,
-            reduce=False,
-        )
-        self.view_args = args
-        self.doc_types = doc_types
-
-    def __iter__(self):
-        args = self.view_args
-        for doc_type in self.doc_types:
-            if args.get("startkey", [None])[0] != doc_type:
-                args.pop("startkey_docid", None)
-                args["startkey"] = [doc_type]
-            args["endkey"] = [doc_type, {}]
-            for result in paginate_view(self.db, **args):
-                yield result['doc']
+    def data_function(**view_kwargs):
+        return db.view('all_docs/by_doc_type', **view_kwargs)
 
-        retried = {}
-        while self.state["retry"] != retried:
-            for doc_id, retries in list(self.state["retry"].iteritems()):
-                if retries == retried.get(doc_id):
-                    continue  # skip already retried (successfully)
-                retried[doc_id] = retries
-                try:
-                    yield self.db.get(doc_id)
-                except ResourceNotFound:
-                    pass
-
-        # save iteration state without offset to signal completion
-        self.state.pop("offset", None)
-        self.state["retry"] = {}
-        self._save_state()
-
-    def retry(self, doc, max_retry=3):
-        """Add document to be yielded at end of iteration
-
-        Iteration order of retry documents is undefined. All retry
-        documents will be yielded after the initial non-retry phase of
-        iteration has completed, and every retry document will be
-        yielded each time the iterator is stopped and resumed during the
-        retry phase. This method is relatively inefficient because it
-        forces the iteration state to be saved to couch. If you find
-        yourself calling this for many documents during the iteration
-        you may want to consider a different retry strategy.
-
-        :param doc: The doc dict to retry. It will be re-fetched from
-        the database before being yielded from the iteration.
-        :param max_retry: Maximum number of times a given document may
-        be retried.
-        :raises: `TooManyRetries` if this method has been called too
-        many times with a given document.
-        """
-        doc_id = doc["_id"]
-        retries = self.state["retry"].get(doc_id, 0) + 1
-        if retries > max_retry:
-            raise TooManyRetries(doc_id)
-        self.state["retry"][doc_id] = retries
-        self._save_state()
+    doc_types = sorted(doc_types)
+    data_function.__name__ = " ".join(doc_types)
 
-    @property
-    def progress_info(self):
-        """Extra progress information
+    view_kwargs = {
+        'limit': chunk_size,
+        'include_docs': True,
+        'reduce': False,
+        'startkey': [doc_types[0]],
```

Comment:
This will raise `IndexError` if the given `doc_types` is empty. I think that should not be an error condition, but instead create an iterator that yields zero items (pretty sure that's how it worked before).